jetty-websocket/websocket-core/src/main/java/org/eclipse/jetty/websocket/core/internal/FrameFlusher.java

```diff
@@ -258,10 +263,12 @@ private boolean succeedEntries()
         for (Entry entry : entries)
         {
             hadEntries = true;
-            notifyCallbackSuccess(entry.callback);
-            entry.release();
+            if(entry.frame.isFin())
+                messagesOut.increment();
             if (entry.frame.getOpCode() == OpCode.CLOSE)
                 endPoint.shutdownOutput();
+            notifyCallbackSuccess(entry.callback);
+            entry.release();
```

Comment:
Is it safe to move the callback to after the endpoint shutdown?

Reply:
after we've sent the close frame we should not be able to send anything else, so i think this is probably better